src/contractor/contractor.cpp

```diff
@@ -494,7 +545,34 @@ std::size_t Contractor::LoadEdgeExpandedGraph(
                 previous_osm_node_id = this_osm_node_id;
             }
 
-            inbuffer.weight = fixed_penalty + new_weight;
+            OSMNodeID from_id;
+            OSMNodeID via_id;
+            OSMNodeID to_id;
+            edge_fixed_penalties_input_stream.read(reinterpret_cast<char *>(&from_id),
+                                                   sizeof(from_id));
+            edge_fixed_penalties_input_stream.read(reinterpret_cast<char *>(&via_id),
+                                                   sizeof(via_id));
+            edge_fixed_penalties_input_stream.read(reinterpret_cast<char *>(&to_id), sizeof(to_id));
+
+            auto turn_iter = turn_penalty_lookup.find(std::make_tuple(from_id, via_id, to_id));
+            if (turn_iter != turn_penalty_lookup.end())
+            {
+                int new_turn_weight = static_cast<int>(turn_iter->second.first * 10);
+
+                if (new_turn_weight + new_weight < original_node_count)
+                {
+                    util::SimpleLogger().Write(logWARNING)
+                        << "turn penalty " << turn_iter->second.first << " for turn " << from_id
+                        << ", " << via_id << ", " << to_id
+                        << " is too negative: clamping turn weight to " << original_node_count;
+                }
+
+                inbuffer.weight = std::max(new_turn_weight + new_weight, original_node_count);
```

Comment:
@TheMarex @daniel-j-h @MoKob @danpat 

I wasn't sure whether to clamp this to `original_node_count` (same as `num_osm_nodes` before it gets modified), `original_node_count * 10`, or `1` (or something else)… 🍝 👾  help?